Arrowgene.Ddon.GameServer/Handler/CraftStartQualityUpHandler.cs

```diff
@@ -0,0 +1,158 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartQualityUpHandler : GameRequestPacketHandler<C2SCraftStartQualityUpReq, S2CCraftStartQualityUpRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartQualityUpHandler));
+
+        private readonly ItemManager _itemManager;
+
+        public CraftStartQualityUpHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = Server.ItemManager;
+        }
+
+        public override S2CCraftStartQualityUpRes Handle(GameClient client, C2SCraftStartQualityUpReq request)
+        {  
+            string equipItemUID = request.ItemUID;
+            var equipItem = Server.Database.SelectStorageItemByUId(equipItemUID);
+            Character character = client.Character;
+            uint pawnid = request.CraftMainPawnID;
+            bool IsGreatSuccess = Random.Shared.Next(5) == 0;
+            string RefineMaterial = request.RefineUID;
+            ushort AddStatusID = request.AddStatusID;
+            byte RandomQuality = 0;
+            int D100 =  Random.Shared.Next(100);
+            List<CDataItemUpdateResult> updateResults;
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new S2CItemUpdateCharacterItemNtc();  
+
+            CDataAddStatusParam AddStat = new CDataAddStatusParam()
```

Comment:
Rewrite this so it looks like:

```c#
List<CDataAddStatusParam> AddStatList = new List<CDataAddStatusParam>()
{
    new CDataAddStatusParam();
};

CDataCurrentEquipInfo CurrentEquipInfo = new CDataCurrentEquipInfo()
{
    ItemUId = equipItemUID,
};
```

Reply:
`CurrentEquipInfo` already has an member of `EquipSlot` allocated with nothing assigned.